.github/workflows/pytest.yml

```diff
@@ -163,6 +164,125 @@ jobs:
           name: Unit Test Results (Python ${{ matrix.python-version }} ${{ matrix.test-markers }})
           path: pytest.xml
 
+  benchmark-pytest:
+    if: >
+      always() && (github.event_name == 'schedule' && github.repository == 'ludwig-ai/ludwig' ||
+      github.event_name == 'pull_request' && github.event.pull_request.merged == true && github.event.pull_request.base.repo.full_name == 'ludwig-ai/ludwig' && !github.event.pull_request.head.repo.fork)
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [3.8]
+        test-markers: ["benchmark"]
+        include:
+          - python-version: 3.8
+            pytorch-version: 1.12.1
+            torchscript-version: 1.10.2
+            ray-version: 2.0.0
+    env:
+      PYTORCH: ${{ matrix.pytorch-version }}
+      MARKERS: ${{ matrix.test-markers }}
+      NEUROPOD_BASE_DIR: "/usr/local/lib/neuropod"
+      NEUROPOD_VERISON: "0.3.0-rc6"
+      TORCHSCRIPT_VERISON: ${{ matrix.torchscript-version }}
+      RAY_VERSION: ${{ matrix.ray-version }}
+
+    name: benchmark, py${{ matrix.python-version  }}, torch-${{ matrix.pytorch-version }}, ray-${{ matrix.ray-version }}, ${{ matrix.os }}
+
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Setup Linux
+        if: runner.os == 'linux'
+        run: |
+          sudo apt-get install -y cmake libsndfile1
+
+      - name: Setup macOS
+        if: runner.os == 'macOS'
+        run: |
+          brew install libuv
+
+      - name: pip cache
```

Comment:
Split the benchmark into its separate workflow `.github/workflows/pytest_benchmarks.yml`. 

I did a quick dive on workflows and I'm not sure they could be helpful in this specific case as the dependencies for the full pytest suite and the benchmarking are different (I also cut down more installation steps in the new workflow file). But I think they could help us better organize the `Install dependencies` step of the `pytest` job in `.github/workflows/pytest.yml` instead of having if conditions. I can do that in a follow up PR.